src/components/SlateEditor/plugins/codeBlock/CodeBlock.jsx

```diff
@@ -0,0 +1,155 @@
+    // men noen data-* attributter (data-format, data-content) er alt definert på embed (???),
+    // derfor lagres dataen som data-code-format og data-code-content i embed taggen
+    model: {
+      code: codeBlock.code || data['code-content'] || '',
+      title:
+        codeBlock.title ||
+        getTitleFromFormat(codeBlock.format || data['code-format']) ||
+        'Text', // TODO tar gjerne i mot forslag for å forbedre denne
+      format: codeBlock.format || data['code-format'] || 'text',
+    },
+    isFirstEdit: data.codeBlock === undefined,
+  };
+};
+
+class CodeBlock extends Component {
+  constructor(props) {
+    super(props);
+    const { isFirstEdit, model } = getInfoFromNode(props.node);
+    this.state = { isFirstEdit, editMode: !model.code, showMenu: false };
+    this.codeBlockRef = createRef();
+    this.toggleEditMode = this.toggleEditMode.bind(this);
+    this.toggleMenu = this.toggleMenu.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.handleRemove = this.handleRemove.bind(this);
+    this.handleUndo = this.handleUndo.bind(this);
+    this.onExit = this.onExit.bind(this);
+  }
+
+  getMenuPosition() {
+    if (this.codeBlockRef.current) {
+      const rect = this.codeBlockRef.current.getBoundingClientRect();
+      return {
+        top: window.scrollY + rect.top + rect.height,
+        left: rect.left,
+      };
+    }
+    return {
+      top: 0,
+      left: 0,
+    };
+  }
+
+  toggleEditMode() {
+    this.setState(prevState => ({ editMode: !prevState.editMode }));
+  }
+
+  toggleMenu() {
+    this.setState(prevState => ({ showMenu: !prevState.showMenu }));
+  }
+
+  handleSave(codeBlock) {
+    const { node, editor } = this.props;
+    const properties = {
+      data: { ...getSchemaEmbed(node), codeBlock: codeBlock },
+    };
+    editor.setNodeByKey(node.key, properties);
+    this.setState({ isFirstEdit: false, editMode: false });
+  }
+
+  handleRemove() {
+    const { editor, node } = this.props;
+    editor.removeNodeByKey(node.key);
+    editor.focus();
+  }
+
+  handleUndo() {
+    const { editor, node } = this.props;
+    editor.unwrapBlockByKey(node.key, 'code-block');
+    editor.focus();
+  }
+
+  onExit() {
+    this.setState(prevState => ({ editMode: false }));
+
+    if (this.state.isFirstEdit) {
+      this.handleUndo();
+    }
+  }
+
+  render() {
+    const { t, editor, node } = this.props;
+    const { editMode, showMenu } = this.state;
+    const { model } = getInfoFromNode(node);
+    const { top, left } = this.getMenuPosition();
+
+    return (
+      <>
+        <span
+          ref={this.codeBlockRef}
+          onClick={this.toggleMenu}
+          onKeyPress={this.toggleMenu}
+          tabIndex={0}
+          role="button">
+          <Codeblock
+            code={model.code}
+            format={model.format}
+            title={model.title}
+          />
+
+          <Portal isOpened={showMenu}>
+            <MathMenu
```

Comment:
Skal forresten denne være her? Eller har den feil navn.

Reply:
Jeg gjenbrukte den komponenten for å redigere/ fjerne blokken, er det en annen komponent for å slette/ endre komponenten jeg bør bruke? 
Ellers bør denne sikkert renames.

Reply:
Aha. Nei du kan fint bruke den, men kanskje døpe den om til BlockMenu eller nokke sånt.